app/services/record_declarations/base.rb

```diff
@@ -1,164 +1,63 @@
 # frozen_string_literal: true
 
-require "abstract_interface"
+require "cpdlp/model"
 
 module RecordDeclarations
   class Base
-    include Participants::ProfileAttributes
-    include AbstractInterface
-    implement_class_method :required_params
-    implement_instance_method :user_profile
+    include CPDLP::Model
 
-    MultipleParticipantDeclarationDuplicate = Class.new(ArgumentError)
-
-    RFC3339_DATE_REGEX = /\A\d{4}-\d{2}-\d{2}T(\d{2}):(\d{2}):(\d{2})([.,]\d+)?(Z|[+-](\d{2})(:?\d{2})?)?\z/i
-
-    attr_accessor :declaration_date, :declaration_type
+    attribute :course_identifier
+    attribute :cpd_lead_provider
+    attribute :declaration_date, :datetime
+    attribute :declaration_type
+    attribute :participant_id
 
+    validates :participant_identity, presence: { message: I18n.t(:invalid_participant, scope: model_name.i18n_key) }
     validates :declaration_date, :declaration_type, presence: true
-    validates :parsed_date, future_date: true, allow_blank: true
-    validate :date_has_the_right_format
-    validate :validate_schedule_present
-    validate :validate_milestone_exists
```

Comment:
did we lose some validations are a result of this change? eg `validate_milestone_exists` seems to have gone?